tests/ui/manual_midpoint.rs

```diff
@@ -0,0 +1,23 @@
+#![warn(clippy::manual_midpoint)]
+
+macro_rules! mac {
+    ($a: expr, $b: expr) => {{ ($a + $b) / 2 }};
+}
+
+fn main() {
+    let a: u32 = 10;
+    let _ = (a + 5) / 2; //~ ERROR: manual implementation of `midpoint`
+
+    let f: f32 = 10.0;
+    let _ = (f + 5.0) / 2.0; //~ ERROR: manual implementation of `midpoint`
+
```

Comment:
Let's also test more complex expressions.
```suggestion

    let _: u32 = 5 + (8 + 8) / 2 + 2; //~ ERROR: manual implementation of `midpoint`
    let _: u32 = const { (8 + 8) / 2 }; //~ ERROR: manual implementation of `midpoint`
    let _: f64 = const { (8.0f64 + 8.) / 2. }; //~ ERROR: manual implementation of `midpoint`
    
    fn zero() -> u32 { 0 }
    let _: u32 = (zero() + zero()) / 2; //~ ERROR: manual implementation of `midpoint`
    
    macro_rules! zeros { () => { 0 } }
    let _: u32 = (zeros!() + zeros!()) / 2; //~ ERROR: manual implementation of `midpoint`
```

Reply:
Why not, but not with `0` which will trigger another warning.

Reply:
In fact, we cannot easily lint this one, as the macro would have been expanded already when we see this. Rewriting the expression would use the expansion, not the macro call.

Reply:
Then the lint must to changed to a multipart suggestion with 3 parts, the first to insert `TYPE::` before the first expression, the second to change ` + ` to `, ` and the last to replace ` / 2` by `` *(nothing)*.

Reply:
I edited my comment, this is now fixed and we get back to the original macro invocation. I've re-instated the test.

Reply:
Great, thanks.